src/main/java/com/intuit/graphql/orchestrator/batch/AuthDownstreamQueryModifier.java

```diff
@@ -232,6 +234,48 @@ public TraversalControl visitFragmentDefinition(
   public TraversalControl visitInlineFragment(InlineFragment node, TraverserContext<Node> context) {
     String typeName = node.getTypeCondition().getName();
     context.setVar(GraphQLType.class, this.graphQLSchema.getType(typeName));
+
+    if(!node.getDirectives(DEFER_DIRECTIVE_NAME).isEmpty()) {
+      Argument deferArg = node.getDirectives(DEFER_DIRECTIVE_NAME).get(0).getArgument(DEFER_IF_ARG);
+      if(graphQLContext.getOrDefault(USE_DEFER, false) && (deferArg == null || ((BooleanValue)deferArg.getValue()).isValue())) {
+        decreaseParentSelectionSetCount(context.getParentContext());
+        return deleteNode(context);
+      } else {
+        //remove directive from query since directive is not built in and will fail downstream if added
+        List<Directive> directives = node.getDirectives()
+                .stream()
+                .filter(directive -> !DEFER_DIRECTIVE_NAME.equals(directive.getName()))
+                .collect(Collectors.toList());
+
+        return changeNode(context, node.transform(builder -> builder.directives(directives)));
+      }
+    }
+
+    return TraversalControl.CONTINUE;
+  }
+
+  @Override
+  public TraversalControl visitFragmentSpread(FragmentSpread node, TraverserContext<Node> context) {
+    context.setVar(GraphQLType.class, this.graphQLSchema.getType(node.getName()));
+
+    if(!node.getDirectives(DEFER_DIRECTIVE_NAME).isEmpty()) {
+      Argument deferArg = node.getDirectives(DEFER_DIRECTIVE_NAME).get(0).getArgument(DEFER_IF_ARG);
+      if(graphQLContext.getOrDefault(USE_DEFER, false) && (deferArg == null || ((BooleanValue)deferArg.getValue()).isValue())) {
+        decreaseParentSelectionSetCount(context.getParentContext());
```

Comment:
If I understood this correctly, USE defer is a feature flag?  If yes, throwing an exception might be a good option as well.  i.e. if feature is not enabled but someone use defer, we can throw an exception